text/0099-otlp-http.md

```diff
@@ -0,0 +1,144 @@
+header if it can receive gzip-encoded responses.
+
+Non-default URL paths for requests MAY be configured on the client and server
+sides.
+
+### Response
+
+#### Success
+
+On success the server MUST respond with `HTTP 200 OK`. Response body MUST be
+ProtoBuf-encoded
+[`ExportTraceServiceResponse`](https://github.com/open-telemetry/opentelemetry-proto/blob/e6c3c4a74d57f870a0d781bada02cb2b2c497d14/opentelemetry/proto/collector/trace/v1/trace_service.proto#L47)
+message for traces and
+[`ExportMetricsServiceResponse`](https://github.com/open-telemetry/opentelemetry-proto/blob/e6c3c4a74d57f870a0d781bada02cb2b2c497d14/opentelemetry/proto/collector/metrics/v1/metrics_service.proto#L44)
+message for metrics.
+
+The server MUST set "Content-Type: application/x-protobuf" response header. If
+the request header "Accept-Encoding: gzip" is present in the request the server
+MAY gzip-encode the response and set "Content-Encoding: gzip" response header.
+
+The server SHOULD respond with success no sooner than after successfully
+decoding and validating the request.
+
+#### Failures
+
+If the processing of the request fails because the request contains data that
+cannot be decoded or is otherwise invalid and such failure is permanent then the
+server MUST respond with `HTTP 400 Bad Request`. The client MUST NOT retry the
+request when it receives `HTTP 400 Bad Request` response.
+
+#### Throttling
+
+If the server receives more requests than the client is allowed or the server is
+overloaded the server SHOULD respond with `HTTP 429 Too Many Requests` or
+`HTTP 503 Service Unavailable` and MAY include "Retry-After" header with a
+recommended time interval in seconds to wait before retrying. The client SHOULD
+honour the waiting interval specified in "Retry-After" header if it is present.
+If the client receives `HTTP 429` or `HTTP 503` response and "Retry-After"
+header is not present in the response then the client SHOULD implement an
+exponential backoff strategy between retries.
+
+#### All Other Responses
+
+All other HTTP responses that are not explicitly listed in this document should
+be treated according to HTTP specification.
+
+If the server disconnects without returning a response the client SHOULD retry
+and send the same request. The client SHOULD implement an exponential backoff
+strategy between retries to avoid overwhelming the server.
+
+### Connection
+
+If the client is unable to connect to the server the client SHOULD retry the
+connection using exponential backoff strategy between retries. The interval
+between retries must have a random jitter.
+
+The client SHOULD keep the connection alive between requests. The client SHOULD
+use "Connection: Keep-Alive" request header.
+
+Server implementations MAY handle OTLP/gRPC and OTLP/HTTP requests on the same
+port and multiplex the connections to the corresponding transport handler based
+on "Content-Type" request header.
```

Comment:
In Jaeger we used sometimes wonky `http/1.1` vs. `http/2` check to separate grpc from http traffic on the same port. Is gRPC going to send something different than `"Content-Type: application/x-protobuf"`? It may be word adding a bit more detail here about how to reuse ports correctly if we're bringing this up at all.

Reply:
Actually, it looks like we switched to context-type as well: https://github.com/jaegertracing/jaeger/blob/46fc5ebe08a90a7fa7d3222f62d0535118a7ba5c/cmd/query/app/server.go#L116

Reply:
Yes, based "Content-Type". gRPC uses "Content-Type: application/grpc", OTLP/HTTP (this proposal) uses "Content-Type: application/x-protobuf".
I can mentioned that "Content-Type" can be used for multiplexing in the doc if you think it is worth it.